recipes/vsg/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+    name = "vsg"
+    description = "VulkanSceneGraph"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.vulkanscenegraph.org"
+    topics = ("vulkan", "scenegraph", "graphics", "3d")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "max_devices": [1,2,3,4],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "max_devices" : 1,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("vulkan-loader/1.3.239.0", transitive_headers=True)
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+
+        if is_msvc_static_runtime(self):
+            raise ConanInvalidConfiguration(f"{self.name} does not support MSVC static runtime (MT/MTd) configurations, only dynamic runtime (MD/MDd) is supported")
+
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+ 
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = False
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["VSG_SUPPORTS_ShaderCompiler"] = 0
+        tc.variables["VSG_MAX_DEVICES"] = self.options.max_devices
+        tc.generate()
+
+        deps = CMakeDeps(self)
+
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE.md", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
```

Comment:
I guess it's a copy/paste from another recipe. Why would there be `.la` files in a CMake build?

Reply:
Good catch, refactored in this [PR](https://github.com/conan-io/conan-center-index/pull/18996)